### PR TITLE
Storage - Enable updates for lifecycle

### PR DESCRIPTION
**Pull request by benrowe1**
#943 Super basic update i did not write any test units but this i think solves the issue.

Many Thanks
Ben

### CODE REVIEW

**Comment by benrowe1** [comment]
I signed it! (Corp Not individual)

**Comment by broady** [comment]
@googlebot check again please

**Comment by benrowe1** [comment]
@googlebot check again please

We were having a lot of issues with the corp CLA keeps saying we have signed but no luck i have signed as an individual for now to push through

**Comment by broady** [comment]
so, possibly bad news. we don't accept pull requests (yet -- i'm working on it.)

@jba, want to review here and then we can move this over to Gerrit?

**Comment by benrowe1** [comment]
np let me know what i can do to help.

File: storage/bucket.go [hunk 1]
```diff
@@ -515,6 +515,8 @@ type BucketAttrsToUpdate struct {
 	// is provided, the value is ignored in the update.
 	CORS []CORS
 
+	Lifecycle Lifecycle
```
Comment (jba) [inline storage/bucket.go]: The problem with this is that we don't whether the user wants to update to this Lifecycle, or not update the Lifecycle at all. So the type should be *Lifecycle.

Comment (benrowe1) [reply storage/bucket.go]: Unfortunately the toRawLifecycle function does not accept pointers and i am just re-using the same code from BucketAttrs. I can change this to a pointer but will need to change this function which will have a knock on effect with BucketAttrs. Let me know how you want me to proceed. I can get the changes done today.

File: storage/bucket_test.go [hunk 2]
```diff
@@ -43,6 +44,15 @@ func TestBucketAttrsToRawBucket(t *testing.T) {
 		MetaGeneration: 39,
 		Created:        time.Now(),
 		Labels:         map[string]string{"label": "value"},
+		Lifecycle: Lifecycle{Rules: []storage.LifecycleRule{storage.LifecycleRule{
```
Comment (jba) [inline storage/bucket_test.go]: Thanks for adding this, it's valuable. But what you really need to do is to add to `TestBucketAttrsToUpdateToRawBucket`.

Comment (benrowe1) [reply storage/bucket_test.go]: Hey Sorry about that was a previous version. I am trying to run the tests following the contrib.md but struggling (appears to keep timing out). Ill finish the tests blind they will need some attention potentially.

**Comment by jadekler** [comment]
Hello! Friendly ping @benrowe1; is this still a change you're interested in?

**Comment by benrowe1** [comment]
This may have now been resolved. I have been moved project and am no longer directly working on this (internally). I will try to confirm if this is still an issue.

**Comment by benrowe1** [comment]
https://github.com/GoogleCloudPlatform/google-cloud-go/commit/96b442738d611f4bdcb5eba1325efa0820783bdd

Looks like the issue was resolved.

**Comment by jadekler** [comment]
Great! Thanks :)

**Pull request closed without merging**